tests/S3/S3ClientTest.php

```diff
@@ -524,6 +531,10 @@ public function testAddsLocationConstraintAutomatically($region, $target, $comma
         } else {
             $this->assertStringNotContainsString($text, $body);
         }
+        //ensures original configuration is not overwritten
+        if ($command === 'CreateBucket') {
```

Comment:
At this point $command is not a string anymore, instead it is an instance of Command. The evaluation here could be :
```php 
if ($command->getName() === 'CreateBucket')
```
Or use another naming strategy to keep the name of the command available till this statement.

Reply:
Also, the assertion here could be more strict and search for the specific node/element. Example:
```php
$this->assertStringContainsString('<Bucket><Type>foo</Type></Bucket>', $body);
```

Reply:
First one is a typo, but sure, I'll add the full xml string